code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/TokenUtils.java

```diff
@@ -65,25 +65,16 @@ public Account getAccountFromJWT() {
     protected abstract String tokenType();
 
     private Map<String, Object> getJsonData(String jwtKey, String tokenType) {
-        if (StringUtils.isEmpty(jwtKey)) {
-            return null;
-        }
         String toParse;
-        String[] tokenArr = jwtKey.split("\\s+");
-        if (tokenArr.length == 2) {
-            if (!StringUtils.equalsIgnoreCase("bearer", StringUtils.trim(tokenArr[0]))) {
-                return null;
-            }
-            toParse = tokenArr[1];
-        } else if (tokenArr.length == 1) {
-            toParse = tokenArr[0];
-        } else {
-            toParse = jwtKey;
-        }
+        toParse = removeBearer(jwtKey);
         String dbJwt = retrieveJwt(toParse);
         if (dbJwt != null){
             toParse = dbJwt;
         }
+        toParse = removeBearer(toParse);
+        if (StringUtils.isEmpty(jwtKey)) {
```

Comment:
I don't understand why this was moved down here.  jwtKey hasn't changed since the start of the function.  The only difference I can see is an exception can be thrown before we reach down here.